src/helper/HelperApp.cpp

```diff
@@ -306,13 +306,17 @@ namespace SDDM {
         if (!authSuccessful) {
 #if defined(Q_OS_LINUX)
             updwtmpx("/var/log/btmp", &entry);
+#elif defined(Q_OS_FREEBSD)
```

Comment:
I don't think this is correct: as commented in a2b14425acc3b5659f44830c3ffd1d8d7c8f4c6a, `pututxline(3)` writes to a file depending on *ut_type* in the entry, and there's no "failued logins" user accounting database. And `pututxline()` has already been called in L301, so at this point it's just writing duplicates.

Reply:
I simply pulled that patch from our port: https://svnweb.freebsd.org/ports/head/x11/sddm/files/patch-src_helper_HelperApp.cpp?view=markup

If this is wrong, should I drop it from the port too?

Reply:
For full context, the code looks like this (after patching in packaging, from L301):

```
        if (!pututxline (&entry))
            qWarning() << "Failed to write utmpx: " << strerror(errno);
        endutxent();
  
        // append to failed login database btmp
        if (!authSuccessful) {
#if defined(Q_OS_FREEBSD)
            pututxline(&entry);
#else
            updwtmpx("/var/log/btmp", &entry);
#endif
        }

        // append to wtmp
        else {
#if defined(Q_OS_FREEBSD)
            pututxline(&entry);
#else
            updwtmpx("/var/log/wtmp", &entry);
#endif
        }
```

On L301, a call to `pututxline(&entry)` .. puts a utx entry. In the subsequent `if()`, there are two more calls to `pututxline(&entry)` which **don't** change the entry .. this is just duplicating entries in the one-and-only utmp database that is used. It's just .. useless, and you could `#if !defined(Q_OS_FREEBSD) ` around the entire `if()` from `// append to failed` to the end of `// append to wtmp` because it's not going to do anything.